cpp/test/sparse/distance.cu

```diff
@@ -73,24 +72,25 @@ class SparseDistanceTest
   {
     make_data();
 
-    dist_config.b_nrows   = params.indptr_h.size() - 1;
-    dist_config.b_ncols   = params.n_cols;
-    dist_config.b_nnz     = params.indices_h.size();
-    dist_config.b_indptr  = indptr.data();
-    dist_config.b_indices = indices.data();
-    dist_config.b_data    = data.data();
-    dist_config.a_nrows   = params.indptr_h.size() - 1;
-    dist_config.a_ncols   = params.n_cols;
-    dist_config.a_nnz     = params.indices_h.size();
-    dist_config.a_indptr  = indptr.data();
-    dist_config.a_indices = indices.data();
-    dist_config.a_data    = data.data();
-
-    int out_size = dist_config.a_nrows * dist_config.b_nrows;
+    int out_size = static_cast<value_idx>(params.indptr_h.size() - 1) *
+                   static_cast<value_idx>(params.indptr_h.size() - 1);
 
     out_dists.resize(out_size, handle.get_stream());
 
-    pairwiseDistance(out_dists.data(), dist_config, params.metric, params.metric_arg);
+    auto out = raft::make_device_matrix_view<value_t, value_idx>(
+      out_dists.data(),
+      static_cast<value_idx>(params.indptr_h.size() - 1),
+      static_cast<value_idx>(params.indptr_h.size() - 1));
+
+    auto x_structure = raft::make_device_compressed_structure_view<value_idx, value_idx, value_idx>(
```

Comment:
This is a mouthfull and one of the reasons I opted to shorten it to `csr_structure` instead of spelling it all out. I wonder if there's another verbiage we can use that might still be consistent with other factory naming conventions but be a little shorter. Not a huge deal, but seeing the function name take up most of this long line of code here is a little concerning.

Reply:
You are right, it is a little verbose. I cannot think of another verbiage off the top of my head right now